crates/router/src/connector/payu/transformers.rs

```diff
@@ -0,0 +1,550 @@
+#[serde(rename_all = "camelCase")]
+pub struct PayuOrderResponseBuyerData {
+    ext_customer_id: Option<String>,
+    email: Option<String>,
+    phone: Option<String>,
+    first_name: Option<String>,
+    last_name: Option<String>,
+    nin: Option<String>,
+    language: Option<String>,
+    delivery: Option<String>,
+    customer_id: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "type")]
+#[serde(rename_all = "UPPERCASE")]
+pub enum PayuOrderResponsePayMethod {
+    #[serde(rename = "CARD_TOKEN")]
+    CardToken,
+    Pbl,
+    Installemnts,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PayuOrderResponseProperty {
+    name: String,
+    value: String,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PayuPaymentsSyncResponse {
+    orders: Vec<PayuOrderResponseData>,
+    status: PayuPaymentStatusData,
+    properties: Option<Vec<PayuOrderResponseProperty>>,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, PayuPaymentsSyncResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            PayuPaymentsSyncResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let order = match item.response.orders.first() {
+            Some(order) => order,
+            _ => Err(errors::ConnectorError::ResponseHandlingFailed)?,
+        };
+        Ok(Self {
+            status: enums::AttemptStatus::from(order.status.clone()),
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(order.order_id.clone()),
+                redirect: false,
+                redirection_data: None,
+                mandate_reference: None,
+            }),
+            amount_captured: Some(order.total_amount.parse::<i64>().unwrap_or_default()),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Serialize)]
+pub struct PayuRefundRequestData {
+    description: String,
+    amount: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct PayuRefundRequest {
+    refund: PayuRefundRequestData,
+}
+
+impl<F> TryFrom<&types::RefundsRouterData<F>> for PayuRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            refund: PayuRefundRequestData {
+                description: item.description.clone().ok_or(
+                    errors::ConnectorError::MissingRequiredField {
+                        field_name: "item.description".to_string(),
+                    },
+                )?,
```

Comment:
Is description a required field?

Reply:
Yes, Description is always required by Payu